packages/core/src/modules/oob/messages/OutOfBandInvitation.ts

```diff
@@ -98,6 +100,15 @@ export class OutOfBandInvitation extends AgentMessage {
     return dids
   }
 
+  // TODO: this only takes into account inline didcomm services, won't work for public dids
+  public getRecipientKeys(): Key[] {
+    return this.services
+      .filter((s): s is OutOfBandDidCommService => typeof s !== 'string')
+      .map((s) => s.recipientKeys)
+      .reduce((acc, curr) => [...acc, ...curr], [])
```

Comment:
What does this reducer do? it seems like a copy without propose.

Reply:
It's basically a flatmap. We have an array of arrays